src/main/java/org/ga4gh/drs/utils/requesthandler/AccessRequestHandler.java

```diff
@@ -0,0 +1,80 @@
+package org.ga4gh.drs.utils.requesthandler;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import org.ga4gh.drs.AppConfigConstants;
+import org.ga4gh.drs.configuration.DrsConfigContainer;
+import org.ga4gh.drs.exception.ResourceNotFoundException;
+import org.ga4gh.drs.model.AccessURL;
+import org.ga4gh.drs.utils.cache.AccessCache;
+import org.ga4gh.drs.utils.cache.AccessCacheItem;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+
+public class AccessRequestHandler implements RequestHandler<AccessURL> {
+
+    @Autowired
+    @Qualifier(AppConfigConstants.MERGED_DRS_CONFIG_CONTAINER)
+    private DrsConfigContainer drsConfigContainer;
+
+    @Autowired
+    private AccessCache accessCache;
+
+    private String objectId;
+    private String accessId;
+
+    public AccessRequestHandler() {
+
+    }
+
+    public AccessURL handleRequest() {
+        AccessCacheItem cacheItem = accessCache.get(objectId + ":" + accessId);
+        if (cacheItem == null) {
+            throw new ResourceNotFoundException("invalid access_id/object_id");
+        }
+
+        AccessURL accessURL = null;
+
+        try {
+            switch (cacheItem.getAccessType()) {
+                case FILE:
+                    accessURL = generateAccessURLForFile(cacheItem.getObjectId(), cacheItem.getAccessId());
+                    break;
+                case HTTPS:
+                    accessURL = null;
+                    break;
+                default:
+                    // TODO THROW ERROR
+                    accessURL = null;
+                    break;
+            }
+        } catch (URISyntaxException e) {
+            // TODO THROW REST CONTROLLER EXCEPTION
+            return null;
+        }
+        
+        return accessURL;
+    }
+
+    private AccessURL generateAccessURLForFile(String objectId, String accessId) throws URISyntaxException {
```

Comment:
The method parameters here are unused. The method calls the `AccessRequestHandler`'s own `getObjectId` and `getAccessId` when it seems like it should be calling the `AccessCacheItem`'s methods or using the passed in parameters.
There's no change in behaviour because the `AccessRequestHandler`'s `objectId` and `accessId` are set to the cache item's ids but this seems like it could be a source of bugs in the future, so either it should use the passed parameters, or not take any

Reply:
agreed, I will update